packages/astroport/src/pair_anchor.rs

```diff
@@ -0,0 +1,195 @@
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+use crate::asset::{Asset, AssetInfo};
+
+use cosmwasm_bignumber::{Decimal256, Uint256};
+use cosmwasm_std::{Binary, Decimal, Uint128};
+use cw20::Cw20ReceiveMsg;
+
+/// The default swap slippage
+pub const DEFAULT_SLIPPAGE: &str = "0.005";
+/// The maximum allowed swap slippage
+pub const MAX_ALLOWED_SLIPPAGE: &str = "0.5";
+
+// Decimal precision for TWAP results
+pub const TWAP_PRECISION: u8 = 6;
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub enum AnchorExecuteMsg {
+    /// ## Description
+    /// Receives a message of type [`Cw20ReceiveMsg`]
+    DepositStable {},
+
+    /// Return stable coins to a user
+    /// according to exchange rate
+    RedeemStable {},
+
+    SetToken(String),
+
+    Receive(Cw20ReceiveMsg),
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub enum AnchorQueryMsg {
+    State { block_height: Option<u64> },
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct StateResponse {
```

Comment:
Do not recreate original Anchor's structs in Astroport code. This way is error prone because it does not track changes on Anchor's side. Import Anchor crate and use their structs instead.

Reply:
@epanchee : Which crate is it?